library-best-practices/JLBP-18.md

```diff
@@ -0,0 +1,49 @@
+[JLBP-18] Use shading sparingly and carefully
+---------------------------------------------
+
+The top answer to the question at
+https://softwareengineering.stackexchange.com/questions/297276/what-is-a-shaded-java-dependency
+does a good job of explaining what shading is. In short, it is a process which
+relocates dependencies so that they don't clash with conflicting versions of the
+same dependencies in a dependency tree, then copies them all into a single jar
+with your own code.
+
+There are a number of problems with shading:
+
+- It can cause considerable size bloat, and if it is applied at multiple layers,
```

Comment:
It can cause considerable bloat. If it is applied at multiple layers,

Reply:
Ok.